### DISCUSSION TITLE
Does anyone know the regex needed to exclude `.story.vue` but include `.vue` files in auto imports?

### DISCUSSION THREAD

**Discussion by Bobakanoosh** [histoire-dev/histoire#252] [General]
I like using [unplugin-vue-components](https://github.com/antfu/unplugin-vue-components) to allow component auto importing.

However, my "default" regex to include vue files is:
`include: [/\.vue$/],`

This ends up including `.story.vue` files as well, which isn't a massive deal, but kind of annoying.

I tried figuring out the regex for it myself with no luck.

**Comment by DaniFoldi** [comment]
Easiest would be to add `exclude: [/\.story\.vue/]` to your config, which `unplugin-vue-components` supports: https://github.com/antfu/unplugin-vue-components#configuration (see last line)

**Comment by alvarosabu** [comment]
You can use `exclude: [/!(*.spec|*.test|*.story).vue]`

**Comment by Bobakanoosh** [comment]
I tried exclude initially and it didn't work. Here is what I have:
```
				include: [/\.vue$/, /\.vue\?vue/],
				exclude: [/\.story\.vue/, /[\\/]node_modules[\\/]/, /[\\/]\.git[\\/]/, /[\\/]\.nuxt[\\/]/],
```

**Comment by 50rayn** [comment]
Hello @Bobakanoosh .
I played a little and suppose that this regexp might work: `^(?!.*(\.story\.).*)(.*)\.\bvue\b$`
